.changeset/heavy-coins-tickle.md

```diff
@@ -0,0 +1,18 @@
+---
+'@shopify/hydrogen': patch
+---
+
+✨ add applyDefault option to createContentSecurityPolicy which allow use to add policy in front of the existing rules instead of overriding them. The default value of applyDefault option is false which is the current behaviour.
```

Comment:
```suggestion
✨ add `applyDefault` option to `createContentSecurityPolicy` which automatically adds Shopify domains to the content security policy, extending whatever rules are passed instead of overriding them. The default value of applyDefault option is false which is the current behavior.
```

Reply:
Actually thinking about this more, I think almost _always_ if the user provides a custom CSP, they will probably also want to keep the Shopify domains as well. So perhaps this should be default to `true`, or inverse it and default it to `false`. Sure it would be a breaking change, but it's relatively minor, and we can make a breaking change for the 2024-01 release. So now would be the time to do it. Thoughts?